test/chunk_info.c

```diff
@@ -544,14 +544,15 @@ test_get_chunk_info_highest_v18(hid_t fapl)
     /* Allocate input (compressed) buffer */
     inbuf = HDmalloc(z_dst_nbytes);
 
-    /* Set chunk size to the compressed chunk size and the chunk point
-       to the compressed data chunk */
-    chunk_size = (hsize_t)z_dst_nbytes;
+    /* zlib-friendly alias for the input buffer */
     z_dst      = (Bytef *)inbuf;
 
     /* Perform compression from the source to the destination buffer */
     ret = compress2(z_dst, &z_dst_nbytes, z_src, z_src_nbytes, aggression);
 
+    /* Set the chunk size to the compressed chunk size */
+    chunk_size = (hsize_t)z_dst_nbytes;
+
```

Comment:
The correct place to set chunk_size is after the call to compress2, when z_dst_nbytes will be set to the size of the compressed data.